zebrad/tests/acceptance.rs

```diff
@@ -1,30 +1,191 @@
-//! Acceptance test: runs the application as a subprocess and asserts its
+//! Acceptance test: runs zebrad as a subprocess and asserts its
 //! output for given argument combinations matches what is expected.
-//!
-//! Modify and/or delete these as you see fit to test the specific needs of
-//! your application.
-//!
-//! For more information, see:
-//! <https://docs.rs/abscissa_core/latest/abscissa_core/testing/index.html>
 
 #![deny(warnings, missing_docs, trivial_casts, unused_qualifications)]
 #![forbid(unsafe_code)]
 
 use abscissa_core::testing::prelude::*;
 use once_cell::sync::Lazy;
 
-/// Executes your application binary via `cargo run`.
+use std::time::Duration;
+
+/// Executes zebrad binary via `cargo run`.
 pub static RUNNER: Lazy<CmdRunner> = Lazy::new(CmdRunner::default);
 
-/*
- * Disabled pending tracing config rework, so that merging abscissa fixes doesn't block on this
- * test failing because there's tracing output.
- *
-/// Example of a test which matches a regular expression
+#[test]
+fn generate_no_args() {
+    let mut runner = RUNNER.clone();
+    let mut cmd = runner.arg("generate").capture_stdout().run();
+    cmd.stdout()
+        .expect_line("# Default configuration values for zebrad.");
+    cmd.wait().unwrap().expect_success();
+}
+
+#[test]
+fn generate_args() {
+    let mut runner = RUNNER.clone();
+    let cmd = runner
+        .args(&["generate", "argument"])
+        .capture_stdout()
+        .run();
+    cmd.wait().unwrap().expect_code(1);
+
+    // Invalid flag
+    let mut runner = RUNNER.clone();
+    let cmd = runner.args(&["generate", "-f"]).capture_stdout().run();
+    cmd.wait().unwrap().expect_code(1);
+
+    // Valid flag but missing argument
+    let mut runner = RUNNER.clone();
+    let cmd = runner.args(&["generate", "-o"]).capture_stdout().run();
+    cmd.wait().unwrap().expect_code(1);
+
+    let mut runner = RUNNER.clone();
+    let cmd = runner
+        .args(&["generate", "-o", "file"])
+        .capture_stdout()
+        .run();
+    cmd.wait().unwrap().expect_success();
+
+    // Todo: Check if the file was created
+}
+
+#[test]
+fn help_no_args() {
+    let mut runner = RUNNER.clone();
+    let cmd = runner.arg("help").capture_stdout().run();
+    cmd.wait().unwrap().expect_success();
+}
+
+#[test]
+fn help_args() {
+    // Invalid argument
+    let mut runner = RUNNER.clone();
+    let cmd = runner.args(&["help", "argument"]).capture_stdout().run();
+    cmd.wait().unwrap().expect_code(1);
+
+    // Invalid flag
+    let mut runner = RUNNER.clone();
+    let cmd = runner.args(&["help", "-f"]).capture_stdout().run();
+    cmd.wait().unwrap().expect_code(1);
+}
+
+#[test]
+fn revhex_no_args() {
+    let mut runner = RUNNER.clone();
+    let _cmd = runner
+        .timeout(Duration::from_secs(1))
+        .arg("revhex")
+        .capture_stdout()
+        .run();
+
+    // Program is waiting for input, we just exit after 1 second
```

Comment:
Issue: please call `wait` to check the exit status:
https://docs.rs/abscissa_core/0.5.2/abscissa_core/testing/process/struct.Process.html#method.wait

We need to call `wait` on every process we launch.